Framework/Algorithms/src/ReflectometryReductionOneAuto3.cpp

```diff
@@ -0,0 +1,1148 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidAlgorithms/ReflectometryReductionOneAuto3.h"
+#include "MantidAPI/BoostOptionalToAlgorithmProperty.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/EnabledWhenProperty.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/MandatoryValidator.h"
+#include "MantidKernel/RegexStrings.h"
+#include "MantidKernel/Strings.h"
+
+#include <boost/lexical_cast.hpp>
+#include <boost/regex.hpp>
+
+namespace Mantid {
+namespace Algorithms {
+
+using namespace Mantid::API;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+namespace { // anonymous
+// Property names
+namespace Prop {
+static const std::string FLIPPERS{"Flippers"};
+static const std::string POLARIZATION_ANALYSIS{"PolarizationAnalysis"};
+} // namespace Prop
+
+namespace CorrectionMethod {
+static const std::string WILDES{"Wildes"};
+static const std::string FREDRIKZE{"Fredrikze"};
+
+// Map correction methods to which correction-option property name they use
+static const std::map<std::string, std::string> OPTION_NAME{
+    {CorrectionMethod::WILDES, Prop::FLIPPERS},
+    {CorrectionMethod::FREDRIKZE, Prop::POLARIZATION_ANALYSIS}};
+
+void validate(const std::string &method) {
+  if (!CorrectionMethod::OPTION_NAME.count(method))
+    throw std::invalid_argument("Unsupported polarization correction method: " +
+                                method);
+}
+} // namespace CorrectionMethod
+
+std::vector<std::string> workspaceNamesInGroup(std::string const &groupName) {
```

Comment:
There seems to be a mixture of east and west `const`, can this be made consistent with the rest of the Reflectometry codebase?

Reply:
This is from v2 so I've added it to #26499